File: ios/Wavepoint/Cleanup/LibraryTrack.swift
import Foundation

enum MusicProvider: String, Equatable, Sendable {
  case spotify
  case appleMusic
}

struct LibraryTrack: Identifiable, Equatable, Sendable {
  let id: String
  let provider: MusicProvider
  let playbackID: String
  let commitID: String
  let title: String
  let artistNames: [String]
  let artworkURL: URL?
  let previewURL: URL?
  let destinationURL: URL?
  let durationMilliseconds: Int
  let addedAt: Date?

  var artistLine: String {
    artistNames.joined(separator: ", ")
  }
}

File: android/app/src/main/java/ai/mapier/swipe/cleanup/LibraryTrack.kt
package ai.mapier.swipe.cleanup

import java.time.Instant

data class LibraryTrack(
  val id: String,
  val playbackId: String,
  val commitId: String,
  val title: String,
  val artistNames: List<String>,
  val artworkUrl: String?,
  val previewUrl: String? = null,
  val destinationUrl: String?,
  val durationMilliseconds: Int,
  val addedAt: Instant?,
) {
  val artistLine: String
    get() = artistNames.joinToString(", ")
}
